### ISSUE TITLE
Bazel null pointer exception when trying to build tensorflow

### ISSUE THREAD

**Issue by alextp** [bazelbuild/bazel#11503]
> ATTENTION! Please read and follow:
> - if this is a _question_ about how to build / test / query / deploy using Bazel, or a _discussion starter_, send it to bazel-discuss@googlegroups.com
> - if this is a _bug_ or _feature request_, fill the form below as best as you can.

### Description of the problem / feature request:

After a long time without building tensorflow I tried to do it and bazel failed with a null pointer exception. I had to install a new version of bazel to do so (bazel 3.0). Going back to bazel 2 and then back to bazel 3 made the issue go away. The stack trace is:

```
FAILED: Build did NOT complete successfully (158 packages loaded, 3519 targets configured)
    currently loading: @bazel_tools//tools/jdk ... (2 packages)
    Fetching @enum34_archive; Restarting.
Internal error thrown during build. Printing stack trace: java.lang.NullPointerException: //tensorflow/tools/pip_package:build_pip_package BuildConfigurationValue.Key[a0f0418923a6f4cd94c6e44c493861009c061da69baa500e98f5248dc6ddeeb6] false
        at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:895)
        at com.google.devtools.build.skyframe.DelegatingWalkableGraph.getDirectDeps(DelegatingWalkableGraph.java:121)
        at com.google.devtools.build.lib.skyframe.SkyframeBuildView.assertSaneAnalysisError(SkyframeBuildView.java:773)
        at com.google.devtools.build.lib.skyframe.SkyframeBuildView.processErrors(SkyframeBuildView.java:616)
        at com.google.devtools.build.lib.skyframe.SkyframeBuildView.configureTargets(SkyframeBuildView.java:454)
        at com.google.devtools.build.lib.analysis.BuildView.update(BuildView.java:404)
        at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.runAnalysisPhase(AnalysisPhaseRunner.java:213)
        at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.execute(AnalysisPhaseRunner.java:124)
        at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:145)
        at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:290)
        at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:95)
        at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:564)
        at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:208)
        at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:603)
        at com.google.devtools.build.lib.server.GrpcServerImpl.lambda$run$2(GrpcServerImpl.java:659)
        at io.grpc.Context$1.run(Context.java:595)
        at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source)
        at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source)
        at java.base/java.lang.Thread.run(Unknown Source)
java.lang.NullPointerException: //tensorflow/tools/pip_package:build_pip_package BuildConfigurationValue.Key[a0f0418923a6f4cd94c6e44c493861009c061da69baa500e98f5248dc6ddeeb6] false
        at com.google.common.base.Preconditions.checkNotNull(Preconditions.java:895)
        at com.google.devtools.build.skyframe.DelegatingWalkableGraph.getDirectDeps(DelegatingWalkableGraph.java:121)
        at com.google.devtools.build.lib.skyframe.SkyframeBuildView.assertSaneAnalysisError(SkyframeBuildView.java:773)
        at com.google.devtools.build.lib.skyframe.SkyframeBuildView.processErrors(SkyframeBuildView.java:616)
        at com.google.devtools.build.lib.skyframe.SkyframeBuildView.configureTargets(SkyframeBuildView.java:454)
        at com.google.devtools.build.lib.analysis.BuildView.update(BuildView.java:404)
        at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.runAnalysisPhase(AnalysisPhaseRunner.java:213)
        at com.google.devtools.build.lib.buildtool.AnalysisPhaseRunner.execute(AnalysisPhaseRunner.java:124)
        at com.google.devtools.build.lib.buildtool.BuildTool.buildTargets(BuildTool.java:145)
        at com.google.devtools.build.lib.buildtool.BuildTool.processRequest(BuildTool.java:290)
        at com.google.devtools.build.lib.runtime.commands.BuildCommand.exec(BuildCommand.java:95)
        at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.execExclusively(BlazeCommandDispatcher.java:564)
        at com.google.devtools.build.lib.runtime.BlazeCommandDispatcher.exec(BlazeCommandDispatcher.java:208)
        at com.google.devtools.build.lib.server.GrpcServerImpl.executeCommand(GrpcServerImpl.java:603)
        at com.google.devtools.build.lib.server.GrpcServerImpl.lambda$run$2(GrpcServerImpl.java:659)
        at io.grpc.Context$1.run(Context.java:595)
FAILED: Build did NOT complete successfully (158 packages loaded, 3519 targets configured)
    currently loading: @bazel_tools//tools/jdk ... (2 packages)
    Fetching @enum34_archive; Restarting.
```

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

It's easy to reproduce on my machine but I don't know how to reproduce on other machines. A coworker said they could reproduce as follows:

```
I managed to repro by running a build, ctrl-C in the middle, run `./configure` again and change a few options
```

### What operating system are you running Bazel on?

Debian

### What's the output of `bazel info release`?

```
$ bazel info release
INFO: Invocation ID: 12b415b1-8bfb-4bd2-9a27-404e4a3c7037
INFO: Options provided by the client:
  Inherited 'common' options: --isatty=1 --terminal_columns=204
INFO: Reading rc options for 'info' from /usr/local/google/home/apassos/tensorflow/.bazelrc:
  Inherited 'common' options: --experimental_repo_remote_exec
INFO: Reading rc options for 'info' from /usr/local/google/home/apassos/tensorflow/.bazelrc:
  Inherited 'build' options: --apple_platform_type=macos --define framework_shared_object=true --define open_source_build=true --java_toolchain=//third_party/toolchains/java:tf_java_toolchain --host_java_toolchain=//third_party/toolchains/java:tf_java_toolchain --define=use_fast_cpp_protos=true --define=allow_oversize_protos=true --spawn_strategy=standalone -c opt --announce_rc --define=grpc_no_ares=true --noincompatible_remove_legacy_whole_archive --noincompatible_prohibit_aapt1 --enable_platform_specific_config --config=v2
INFO: Reading rc options for 'info' from /usr/local/google/home/apassos/tensorflow/.tf_configure.bazelrc:
  Inherited 'build' options: --action_env PYTHON_BIN_PATH=/usr/bin/python3 --action_env PYTHON_LIB_PATH=/usr/lib/python3/dist-packages --python_path=/usr/bin/python3 --config=xla --action_env TF_CONFIGURE_IOS=0
INFO: Reading rc options for 'info' from /usr/local/google/home/apassos/.bazelrc:
  Inherited 'build' options: --disk_cache=/usr/local/google/home/apassos/bazel_cache --experimental_strict_action_env
INFO: Found applicable config definition build:v2 in file /usr/local/google/home/apassos/tensorflow/.bazelrc: --define=tf_api_version=2 --action_env=TF2_BEHAVIOR=1
INFO: Found applicable config definition build:xla in file /usr/local/google/home/apassos/tensorflow/.bazelrc: --action_env=TF_ENABLE_XLA=1 --define=with_xla_support=true
INFO: Found applicable config definition build:linux in file /usr/local/google/home/apassos/tensorflow/.bazelrc: --copt=-w --define=PREFIX=/usr --define=LIBDIR=$(PREFIX)/lib --define=INCLUDEDIR=$(PREFIX)/include --cxxopt=-std=c++14 --host_cxxopt=-std=c++14 --config=dynamic_kernels
INFO: Found applicable config definition build:dynamic_kernels in file /usr/local/google/home/apassos/tensorflow/.bazelrc: --define=dynamic_loaded_kernels=true --copt=-DAUTOLOAD_DYNAMIC_KERNELS
release 3.0.0
```

### What's the output of `git remote get-url origin ; git rev-parse master ; git rev-parse HEAD` ?

```
 git remote get-url origin ; git rev-parse master ; git rev-parse HEAD
git@github.com:tensorflow/tensorflow.git
1de7105aeb3358a290f09c3ee46c5fe760a90c75
1de7105aeb3358a290f09c3ee46c5fe760a90c75
```

###  Have you found anything relevant by searching the web?

No

**Comment by hlopko** [comment]
@lberki look bazel crash :) (nerd-snipe attempt)

**Comment by ulfjack** [comment]
I seem to remember having seen that stack trace before...

Yeah, I think it's #11166. Do you have a repro?

**Comment by janakdr** [comment]
Closed by de6808488a5b97fd154fdb5dcdf28f030e60dbb8
